src/common/ini_handler.c

```diff
@@ -65,21 +94,136 @@ uint8_t ini_save_file(const char *ini_save_str) {
     return 1;
 }
 
-uint8_t ini_load_file(void *user_struct) {
-    UINT written_bytes = 0;
-    FIL ini_file;
-    ini_file_str_t ini_str;
+buddy_ini_error ini_load_file(void *user) {
+    FIL fileObject;
+    buddy_ini_error error = BUDDY_INI_OK;
+    char line[INI_MAX_LINE];
+    char *start;
+    char *end;
+    char *name;
+    char *value;
+    int lineno = 0;
+    uint8_t skip_next_read = 0;
+
+    char section[MAX_SECTION] = "";
+    char prev_name[MAX_NAME] = "";
+
+    // read line by line and parse it
+    FRESULT res = f_open(&fileObject, ini_file_name, FA_READ);
+    if (res) {
+        f_close(&fileObject);
+        return BUDDY_INI_FILE_READ;
+    }
 
-    uint8_t file_init = f_open(&ini_file, network_ini_file_name, FA_READ);
-    uint8_t file_read = f_read(&ini_file, &ini_str, MAX_INI_SIZE, &written_bytes);
-    uint8_t file_close = f_close(&ini_file);
+    /* Scan through stream line by line */
+    while ((f_eof(&fileObject) == 0) && (f_error(&fileObject) == 0)) {
+        if (NULL == f_gets(line, INI_MAX_LINE, &fileObject)) {
+            f_close(&fileObject);
+            return BUDDY_INI_FILE_READ;
+        }
+        // skip the processing if it is part of previous line
+        if (skip_next_read) {
+            skip_next_read = 0;
+            continue;
+        }
+        // check if the line was completely read
+        if (NULL == strchr(line, '\n')) {
+            skip_next_read = 1;
+        }
 
-    if (file_init || file_read || file_close) {
-        return 0;
-    }
+#if INI_ALLOW_REALLOC && !INI_USE_STACK
```

Comment:
Was this piece of code copied from the `inih` library? There are multiple unnecessary #defines like INI_ALLOW_REALLOC and INI_USE_STACK. 

Of course we want to avoid allocating dynamic memory and yes, we'd like to do the stuff on stack only.

Reply:
please have a look at the eeprom.c/.h file changes, the existing variant based function had issue with reading writing string to EEPROM

Reply:
When you copied a piece of code from the INIH library (which is weird and suspicious on its own), please prune the meaningless compile-time options.

Reply:
I had strategy to keep the whole thing which worked, if you remove something it could have some influence.